### PR TITLE
Feat/admin command

### PR DESCRIPTION
**Pull request by topocount**
implement user-level service via the CLI that provides a browser frontend for interacting with the ledger and cred explorer. Endpoints are also implemented for consuming the ledger.json file and writing it back to disk via a GET and POST commands, respectively.

test-plan: run `sourcecred site`, then `sourcecred admin` to start the service, then
post some json to the path by doing something like
```
$ curl --header "Content-Type: application/json" \
  --request POST \
  --data '{"username":"derp","password":"slerp"}' \
http://localhost:6006/data/ledger.json
```
and ensure there are no errors.

then
```
$ curl http://localhost:6006/data/ledger.json
```
and ensure the submitted json is returned

Additionally, browse to `http://localhost:6006` and ensure the cred explorer and ledger identity editor load

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
     "history": "^4.10.1",
     "htmlparser2": "^4.1.0",
     "isomorphic-fetch": "^2.2.1",
+    "json-server": "^0.16.1",
```
Comment (decentralion) [inline package.json:32]: Unused dep

File: package.json [hunk 2]
```diff
@@ -48,6 +49,7 @@
     "remove-markdown": "^0.3.0",
     "retry": "^0.12.0",
     "rimraf": "^3.0.2",
+    "serve-static": "^1.14.1",
```
Comment (decentralion) [inline package.json:52]: Unused dep

File: src/cli/admin.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  // for fetching the ledger.json
+  server.use("/data", express.static(path.join(__dirname, "data")));
+  server.use(express.static("."));
+
+  // middleware that parses json for us
+  server.use(express.json());
+  // write posted ledger.json files to disk
+  server.post("/data/ledger.json", (req, res) => {
+    console.log(req.body);
```
Comment (decentralion) [inline src/cli/admin.js:29]: remove console.log

File: src/cli/admin.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    const stringifiedLedger = JSON.stringify(req.body);
```
Comment (decentralion) [inline src/cli/admin.js:30]: It should get stringified by the frontend.

File: src/cli/admin.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    try {
+      fs.writeFileSync("./data/ledger.json", stringifiedLedger, "utf8");
+    } catch (e) {
+      res.status(500).send(`error saving ledger.json file: ${e}`);
+    }
+    res.status(201).end();
+  });
+
+  server.listen(6006, () => {
+    console.info("Server is running");
```
Comment (decentralion) [inline src/cli/admin.js:40]: change to:

`admin server running: navigate to http://localhost:6006`

File: flow-typed/npm/express_v4.17.x.js [hunk 6]
```diff
@@ -323,6 +323,14 @@ declare type express$UrlEncodedOptions = {
...
 }
 
+declare type express$TextOptions = {
+  defaultCharset?: string,
+  inflate?: boolean,
+  limit?: mixed,
+  type?: mixed,
+  verify?: Function,
+}
+
```
[the remark this answers is not in the document]
Comment (topocount) [reply flow-typed/npm/express_v4.17.x.js:333]: It's manual. no further changes from upstream

File: src/cli/admin.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+  const server = express();
+
+  // for fetching data/ledger.json
+  server.use("/data", express.static(path.join(__dirname, "data")));
+  // serve the static admin site
+  server.use(express.static("."));
+
+  // middleware that parses text request bodies for us
+  server.use(express.text());
```
[the remark this answers is not in the document]
Comment (topocount) [reply src/cli/admin.js:27]: This enables express to parse plaintext request bodies: https://expressjs.com/en/4x/api.html#express.text

Express doesn't know how to parse even the simplest request bodies without loading the appropriate middleware

[the remark this answers is not in the document]
Comment (topocount) [reply src/cli/admin.js:27]: to be more explicit. the request body is returned to us as `undefined` without this line

**Pull request merged**